addons/mass_mailing_sms/views/mailing_mailing_views.xml

```diff
@@ -1,9 +1,23 @@
 <?xml version="1.0" encoding="utf-8"?>
 <odoo><data>
-    <record id="mailing_mailing_view_form" model="ir.ui.view">
+
+    <record model="ir.ui.view" id="mailing_mailing_view_search_sms">
+        <field name="name">mailing.mailing.search</field>
+        <field name="model">mailing.mailing</field>
+        <field name="inherit_id" ref="mass_mailing.view_mail_mass_mailing_search"/>
+        <field name="mode">primary</field>
+        <field name="arch" type="xml">
+            <xpath expr="//filter[@name='assigned_to_me']" position="attributes">
+                <attribute name="string">My SMS Marketing</attribute>
+            </xpath>
+        </field>
+    </record>
+
+    <record id="mailing_mailing_view_form_sms" model="ir.ui.view">
         <field name="name">mailing.mailing.view.form.inherit.sms</field>
         <field name="model">mailing.mailing</field>
         <field name="inherit_id" ref="mass_mailing.view_mail_mass_mailing_form"/>
+        <field name="mode">primary</field>
```

Comment:
I still vote to remove the primary mode as it breaks activities (notably). Having different views on same model is broken by design in a lot of cases. Do you really need that ?